xbmc/cores/VideoPlayer/VideoPlayer.h

```diff
@@ -168,27 +168,20 @@ class CCurrentStream
   }
 };
 
-typedef struct SelectionStream
+class SelectionStream : public DemuxSelectInfo
```

Comment:
uff, you make my life not easier :-)

I used it exactly in Demux because I know that you don't like circular dependencies.
Just rename it? Remove the "Selection" out of it?